scripts/plot_precision.py

```
@@ -1,181 +1,364 @@
+        "grid.alpha": 0.3,
+        "savefig.bbox": "tight",
+    })
+    if use_latex:
+        plt.rcParams["text.usetex"] = True
+
+
+def type_color_map(types: Iterable[str]) -> dict[str, tuple]:
+    """Stable color assignment per arithmetic type.
+
+    Using `tab10` gives 10 distinct colors at the start and then cycles —
+    fine for the 5–8 types this sweep typically produces. Sorting the
```

Ruff (0.15.10)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)